dunwoody_disaster/views/fightScreen.py

```diff
@@ -21,172 +22,105 @@
-
-        self.mainLayout.addItem(
-            QSpacerItem(
-                40,
-                40,
-                QSizePolicy.Policy.MinimumExpanding,
-                QSizePolicy.Policy.MinimumExpanding,
-            ),
-            row,
-            rightCol + 3,
-        )
-        colm += 1
-
-        #############################################################
-        self.P2Arsenal = Arsenal(
-            self.p2_selector, self.player2.weapons, self.player2.defenses
-        )
-        self.mainLayout.addWidget(self.P2Arsenal, 1, colm, 16, 1)
-
-        colm += 1
-        self.mainLayout.addItem(
-            QSpacerItem(30, 50, QSizePolicy.Policy.Fixed, QSizePolicy.Policy.Fixed),
-            row,
-            colm,
-        )
-
-        self.timer.start(2000)
-        self.timer.timeout.connect(self.Fight)
-
-    def SetFightFlag(self):
-        if self.CanFight(self.p1_selector) and self.CanFight(self.p2_selector):
-            self.fightFlag = True
-        else:
-            print("You must select 2 actions to fight!")
-
-    def CanFight(self, actionSelector: ActionSelector):
-        return (actionSelector.attack and actionSelector.defense) is not None
-
-    def Fight(self):
-        if self.fightFlag:
-            self.fight_Btn.setEnabled(False)
-            self.player1, self.player2 = self.fightSequence.Fight(
-                self.p1_selector,
-                self.p2_selector,
-            )
-            self.player1.set_health(self.player1.curHealth)
-            self.player1.set_magic(self.player1.curMagic)
-            self.player1.set_stamina(self.player1.curStamina)
-
-            self.player2.set_health(self.player2.curHealth)
-            self.player2.set_magic(self.player2.curMagic)
-            self.player2.set_stamina(self.player2.curStamina)
-            if self.player1.curHealth <= 0 or self.player2.curHealth <= 0:
-                self.doneFlag = True
-                self.timer.stop()
-                if self.player1.curHealth == 0:
-                    print("Player 2 Wins!")
-                else:
-                    self.onWin()
-            self.fightFlag = False
-            self.fight_Btn.setEnabled(True)
-            self.fightFlag = False
-            self.fight_Btn.setEnabled(True)
-
-    def onWin(self):
-        vc = VictoryScreen(self.fightSequence)
-        self.stacked_layout.addWidget(vc)
-        self.stacked_layout.setCurrentWidget(vc)
+        layout = QHBoxLayout()
+        layout.setSpacing(0)
+        layout.setContentsMargins(0, 0, 0, 0)
+        self.setLayout(layout)
+
+        layout.addItem(DD.expander(True, False, 0))
+
+        arsenal = Arsenal(self.p1_selector, self.player1.weapons, self.player1.defenses)
+        layout.addWidget(arsenal)
+
+        layout.addItem(DD.expander(True, False))
+        # layout.addLayout(self.center_layout(), stretch=1)
+        layout.addLayout(self.center_layout())
+        layout.addItem(DD.expander(True, False))
```

Comment:
@instMeyFTW question that you might be able to answer:
I have these `expander`'s which are the `MinimumExpanding` spacers you showed me earlier. The problem is that I have an expander inside of this `center_layout` function, but it doesn't expand unless [stretch](https://doc.qt.io/qt-6/qboxlayout.html#addLayout) is set. But then, if stretch is set, the expanders on line 50 & 53 don't stretch anymore.

If you want to checkout this branch and see if you can get it to work be my guest. the expander I'm trying to get to work is the one in-between the character pictures on the fight screen.